From e0b5eda908b40bbf0ae70940f7c7e4e90e63f807 Mon Sep 17 00:00:00 2001
From: Sai Vemuri <svemuri@redhat.com>
Date: Thu, 11 Sep 2014 22:43:54 -0400
Subject: [ethernet] cxgb4: Fixed incorrect check for memory operation in t4_memory_rw

Message-id: <1410475447-94392-96-git-send-email-svemuri@redhat.com>
Patchwork-id: 93800
O-Subject: [RHEL7.1 PATCH BZ 1124947 095/108] cxgb4: Fixed incorrect check for memory operation in t4_memory_rw
Bugzilla: 1124947
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

Fix incorrect check introduced in commit fc5ab020 ("cxgb4: Replaced the
backdoor mechanism to access the HW memory with PCIe Window method"). We where
checking for write operation and doing a read, changed it accordingly.

Signed-off-by: Hariprasad Shenai <hariprasad@chelsio.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
(cherry picked from commit c81576c225b942496feb9fbef5bca506bc64ce57)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/chelsio/cxgb4/t4_hw.c b/drivers/net/ethernet/chelsio/cxgb4/t4_hw.c
index d25ba4e..8ff46cf 100644
--- a/drivers/net/ethernet/chelsio/cxgb4/t4_hw.c
+++ b/drivers/net/ethernet/chelsio/cxgb4/t4_hw.c
@@ -546,7 +546,7 @@ int t4_memory_rw(struct adapter *adap, int win, int mtype, u32 addr,
   unsigned char *bp;
   int i;
 
-  if (dir == T4_MEMORY_WRITE) {
+  if (dir == T4_MEMORY_READ) {
    last.word = (__force __be32) t4_read_reg(adap,
        mem_base + offset);
    for (bp = (unsigned char *)buf, i = resid; i < 4; i++)
-- 
1.7.1